sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/AzureBlobLease.cs

```diff
@@ -11,48 +12,66 @@ class AzureBlobLease : Lease
 	{
         readonly bool isOwned;
 
-	    // ctor needed for deserialization
-	    internal AzureBlobLease()
+        readonly TimeSpan LeaseDuration;
+
+        private bool IsInfiniteLease => this.LeaseDuration > TimeSpan.FromSeconds(60);
+
+        // ctor needed for deserialization
+        internal AzureBlobLease()
 		{
 		}
 
-	    internal AzureBlobLease(string partitionId, CloudBlockBlob blob) : base(partitionId)
+	    internal AzureBlobLease(string partitionId, CloudBlockBlob blob, TimeSpan LeaseDuration) : base(partitionId)
```

Comment:
this parameter name does not follow our casing  conventions. It should be camel Cased.